packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1896,8 +1922,222 @@ export class MergeTree {
 			segmentGroup: undefined,
 		};
 
-		const { segment: startSeg } = this.getContainingSegment(start, refSeq, clientId);
-		const { segment: endSeg } = this.getContainingSegment(end - 1, refSeq, clientId);
+		const { segment: startSeg } =
+			startPlace.pos === -1
+				? { segment: startPlace.side === Side.After ? this.startOfTree : this.endOfTree }
+				: this.getContainingSegment(startPlace.pos, refSeq, clientId);
+		const { segment: endSeg } =
+			endPlace.pos === -1
+				? { segment: endPlace.side === Side.After ? this.startOfTree : this.endOfTree }
+				: this.getContainingSegment(endPlace.pos, refSeq, clientId);
+		assert(
+			startSeg !== undefined && endSeg !== undefined,
+			0xa3f /* segments cannot be undefined */,
+		);
+
+		obliterate.start = this.createLocalReferencePosition(
+			startSeg,
+			startPlace.side === Side.Before ? 0 : Math.max(startSeg.cachedLength - 1, 0),
+			ReferenceType.StayOnRemove,
+			{
+				obliterate,
+			},
+		);
+
+		obliterate.end = this.createLocalReferencePosition(
+			endSeg,
+			endPlace.side === Side.Before ? 0 : Math.max(endSeg.cachedLength - 1, 0),
+			ReferenceType.StayOnRemove,
+			{
+				obliterate,
+			},
+		);
+
+		const markMoved = (
+			segment: ISegment,
+			pos: number,
+			_start: number,
+			_end: number,
+		): boolean => {
+			if (
+				(startPlace.side === Side.After && startPos === pos + segment.cachedLength) || // exclusive start segment
+				(endPlace.side === Side.Before &&
+					endPos === pos &&
+					isSegmentPresent(segment, { refSeq, localSeq })) // exclusive end segment
+			) {
+				// We walk these segments because we want to also walk any concurrently inserted segments between here and the obliterated segments.
+				// These segments are outside of the obliteration range though, so return true to keep walking.
+				return true;
+			}
+			const existingMoveInfo = toMoveInfo(segment);
+			// TODO: if segment was inserted inside of a later local obliteration range
+			// then it shouldn't be considered obliterated for other clients
+			// for refSeqs between the insertion's seq and the obliterate's seq.
+
+			if (segment.prevObliterateByInserter?.seq === UnassignedSequenceNumber) {
```

Comment:
is this condition ever hit? I don't think the local client will ever run this method for previous obliterates

Reply:
The condition is hit, I added a line to the comment explaining the circumstances where it is hit a bit more.